packages/contracts/evm-contracts/contracts/token/InverseBaseProjected1155.sol

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import {Strings} from "@openzeppelin/contracts/utils/Strings.sol";
+import {ERC1155} from "@openzeppelin/contracts/token/ERC1155/ERC1155.sol";
+import {ERC1155Supply} from "@openzeppelin/contracts/token/ERC1155/extensions/ERC1155Supply.sol";
+import {IERC1155MetadataURI} from "@openzeppelin/contracts/token/ERC1155/extensions/IERC1155MetadataURI.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {IERC165} from "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+import {IInverseProjected1155} from "./IInverseProjected1155.sol";
+import {IInverseBaseProjected1155} from "./IInverseBaseProjected1155.sol";
+import {IUri} from "./IUri.sol";
+
+/// @dev A Paima Inverse Projection ERC1155 token where initialization is handled by the base-layer.
+/// A standard ERC1155 that accepts calldata in the mint function for any initialization data needed in a Paima dApp.
+contract InverseBaseProjected1155 is IInverseBaseProjected1155, ERC1155Supply, Ownable {
+    using Strings for uint256;
+
+    string public name;
+    string public symbol;
+
+    /// @dev The token ID that will be minted when calling the `mint` function.
+    uint256 public currentTokenId;
+    /// @dev Base URI that is used in the `uri` function to form the start of the token URI.
+    string public baseURI;
+    /// @dev Base extension that is used in the `uri` function to form the end of the token URI.
+    string public baseExtension;
+
+    /// @dev Sets the NFT's `name`, `symbol`, and transfers ownership to `owner`.
+    /// Also sets `currentTokenId` to 1.
+    constructor(
+        string memory _name,
+        string memory _symbol,
+        address _owner
+    ) ERC1155("") Ownable(_owner) {
+        name = _name;
+        symbol = _symbol;
+        currentTokenId = 1;
+    }
+
+    /// @dev Returns true if this contract implements the interface defined by `interfaceId`. See EIP165.
+    function supportsInterface(
+        bytes4 interfaceId
+    ) public view virtual override(IERC165, ERC1155) returns (bool) {
+        return
+            interfaceId == type(IInverseProjected1155).interfaceId ||
+            interfaceId == type(IInverseBaseProjected1155).interfaceId ||
+            super.supportsInterface(interfaceId);
+    }
+
+    /// @dev Mints `value` of a new token to transaction sender, passing `initialData` to be emitted in the event.
+    /// Increases the `currentTokenId`.
+    /// Reverts if transaction sender is a smart contract that does not implement IERC1155Receiver-onERC1155Received.
+    /// Emits the `Minted` event.
+    /// Returns the id of the minted token.
+    function mint(
+        uint256 value,
+        bytes memory data,
+        string calldata initialData
```

Comment:
although it's clear the difference between `data` and `initialData` if you're familiar with both ERC1155 and Paima, probably we should mention something in the comment since it may not be obvious for others (I noticed we put a comment to explain this in the app-initiated variation, but not this one)

Reply:
comment added